modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockRelations.scala

```diff
@@ -0,0 +1,32 @@
+package org.tessellation.dag.l1.domain.block
+
+import cats.effect.kernel.Async
+import cats.syntax.eq._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.{BlockReference, transaction}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+object BlockRelations {
+  def areRelated[F[_]: Async: KryoSerializer](block: Hashed[DAGBlock]): F[Signed[DAGBlock] => Boolean] =
+    areRelated(Set(block), Set.empty)
+
+  def areRelated[F[_]: Async: KryoSerializer](
+    blocks: Set[Hashed[DAGBlock]],
+    blocksRefs: Set[BlockReference]
+  ): F[Signed[DAGBlock] => Boolean] =
+    for {
+      transactionsRef <- blocks.toList
+        .flatTraverse(_.transactions.toSortedSet.toList.traverse(txn => transaction.TransactionReference.of(txn)))
+        .map(_.toSet)
+      dstAddresses = blocks.flatMap(_.transactions.toSortedSet.toList.map(_.value.destination))
+    } yield
+      (isRelated: Signed[DAGBlock]) =>
```

Comment:
I would try to figure out a better name. `isRelated` sounds like it's a boolean whereas it's a block that's being checked 🤔

Reply:
BlockRelations.areRelated is also weird/funny name :D